web-demo-app/src/app/board/submodule.ts

```diff
@@ -0,0 +1,30 @@
+import { BoardModule } from './board.module';
+import { Credentials } from '../credentials';
+import { Session } from '@closerplatform/closer-sdk';
+import { SpinnerClient } from '@swagger/spinner';
+
+export abstract class SubModule {
+  public abstract readonly NAME: string;
+  protected inner: JQuery;
+
+  constructor (public boardModule: BoardModule, public credentials: Credentials) { }
+
+  public abstract init(session: Session, spinnerClient: SpinnerClient): Promise<void>;
```

Comment:
Tego chyba też nie potrzebujemy

Reply:
to dla upewnienia się, że ta metoda będzie w klasach dziedziczących, w boardModule, używamy init na obiektach klasy SubModule